pkg/porter/publish.go

```diff
@@ -61,13 +100,61 @@ func (p *Porter) Publish(opts PublishOptions) error {
 		return err
 	}
 
-	return p.Registry.PushBundle(bun, p.Manifest.BundleTag, opts.InsecureRegistry)
+	return p.Registry.PushBundle(bun, tag, opts.InsecureRegistry)
+}
+
+// TODO: tests!
+func (p *Porter) publishFromArchive(opts PublishOptions) error {
+	fmt.Fprintf(p.Out, "Extracting bundle from archive %s...\n", opts.ArchiveFile)
+	source, err := filepath.Abs(opts.ArchiveFile)
+	if err != nil {
+		return errors.Wrapf(err, "could not determine absolute path to archive file %s", opts.ArchiveFile)
+	}
+
+	tmpDir, err := p.FileSystem.TempDir("", "porter")
+	if err != nil {
+		return errors.Wrap(err, "error creating temp directory for archive extraction")
+	}
+	defer p.FileSystem.RemoveAll(tmpDir)
+
+	l := loader.NewLoader()
+	// as of writing, NewImporter always returns nil as error
+	imp, _ := packager.NewImporter(source, tmpDir, l, false)
+
+	err = imp.Import()
+	if err != nil {
+		return errors.Wrapf(err, "failed to extract bundle from archive %s", opts.ArchiveFile)
+	}
+
+	// TODO: do we want to import bundle into Porter's cache?
```

Comment:
We have another open issue to have special handling of the cache and cache invalidation on publish, #678, so I suggest maybe leaving that for a follow on, or as part of that issue.